src/helpers/class-components.php

```
@@ -0,0 +1,70 @@
+<?php
+/**
+ * Helpers for components
+ *
+ * @package Eightshift_Libs\Helpers
+ */
+
+declare( strict_types=1 );
+
+namespace Eightshift_Libs\Helpers;
+
+/**
+ * Helpers for components
+ */
+class Components {
+
+  /**
+   * Makes sure the output is string. Useful for converting an array of components into a string.
+   *
+   * @param  array|string $array_or_string Variable we need to convert into a string.
+   * @return string
+   */
+  public static function ensure_string( $array_or_string ) : string {
+    $output = '';
+
+    if ( is_array( $array_or_string ) ) {
+      $output = implode( '', $array_or_string );
+    } elseif ( is_string( $array_or_string ) ) {
+      $output = $array_or_string;
+    }
+
+    return $output;
+  }
```

What happens when you pass something that is not `array` or `string`? Like integer?

---

Hmm, yeah didn't thought about that, I'll make it throw an exception in that case.

---

Or you could have used `strval()`?

https://www.php.net/manual/en/function.strval.php

---

Sure I could cast it to string but I'd rather if it errored out with a helpful msg on invalid use rather than the function trying to make it work or silently failing.

It's the same reasoning why the `render()` method below does the same thing (our previous implementations were always "If the template doesn't exist, just do nothing, a.k.a silently fail"). Debugging that is hard.